lib/lockfile-parser.ts

```diff
@@ -20,23 +21,32 @@ import {
 import 'core-js/features/object/entries';
 
 export default class LockfileParser {
-  public static readFileSync(path: string): LockfileParser {
-    const fileContents = fs.readFileSync(path, 'utf8');
-    return this.readContents(fileContents);
+  public static readFileSync(lockfilePath: string): LockfileParser {
+    const fileContents = fs.readFileSync(lockfilePath, 'utf8');
+    const rootName = path.basename(path.dirname(path.resolve(lockfilePath)));
+    return this.readContents(fileContents, {
+      name: rootName,
+      version: '0.0.0',
+    });
   }
 
-  public static readContents(contents: string): LockfileParser {
-    return new LockfileParser(yaml.safeLoad(contents));
+  public static readContents(
+    contents: string,
+    rootPkgInfo?: PkgInfo
+  ): LockfileParser {
+    return new LockfileParser(yaml.safeLoad(contents), rootPkgInfo);
   }
 
+  private rootPkgInfo: PkgInfo | undefined = undefined;
   private internalData: Lockfile;
 
-  public constructor(hash: Lockfile) {
+  public constructor(hash: Lockfile, rootPkgInfo?: PkgInfo) {
```

Comment:
TS has shorthand for this:
```ts
public constructor(hash: Lockfile, private rootPkgInfo?: PkgInfo) {
```
And then you can remove
```ts
private rootPkgInfo: PkgInfo | undefined = undefined;
```
Not everyone enjoys that syntax though though :D

Reply:
TIL, indeed not quite sold on it being better tho yet.